app/main/views/suppliers.py

```diff
@@ -496,24 +496,62 @@ def move_user_to_new_supplier(supplier_id):
 @main.route('/suppliers/<int:supplier_id>/services', methods=['GET'])
 @role_required('admin', 'admin-ccs-category')
 def find_supplier_services(supplier_id):
+    remove_services_for_framework_slug = request.args.get('remove', None)
+
+    frameworks = data_api_client.find_frameworks()['frameworks']
+    supplier = data_api_client.get_supplier(supplier_id)["suppliers"]
+    services = data_api_client.find_services(supplier_id)['services']
 
-    frameworks = data_api_client.find_frameworks()
-    supplier = data_api_client.get_supplier(supplier_id)
-    services = data_api_client.find_services(supplier_id)
     frameworks_services = {
         framework_slug: list(framework_services)
         for framework_slug, framework_services in
-        groupby(sorted(services['services'], key=itemgetter('frameworkSlug')), key=itemgetter('frameworkSlug'))
+        groupby(sorted(services, key=itemgetter('frameworkSlug')), key=itemgetter('frameworkSlug'))
     }
 
+    if remove_services_for_framework_slug:
+        if remove_services_for_framework_slug not in frameworks_services:
+            abort(400, 'Invalid framework')
+        if not any(i['status'] == 'published' for i in frameworks_services[remove_services_for_framework_slug]):
+            abort(400, 'Invalid framework')
+
+        remove_services_for_framework = next(filter(
+            lambda i: i['slug'] == remove_services_for_framework_slug,
+            frameworks
+        ))
+    else:
+        remove_services_for_framework = None
+
     return render_template(
-        "view_supplier_services.html",
-        frameworks=frameworks['frameworks'],
+        'view_supplier_services.html',
+        frameworks=frameworks,
         frameworks_services=frameworks_services,
-        supplier=supplier["suppliers"]
+        supplier=supplier,
+        remove_services_for_framework=remove_services_for_framework,
     )
 
 
+@main.route('/suppliers/<int:supplier_id>/services', methods=['POST'])
+@role_required('admin', 'admin-ccs-category')
+def disable_supplier_services(supplier_id):
+    remove_services_for_framework = request.args.get('remove')
+    if not remove_services_for_framework:
+        abort(400, 'Invalid framework')
+
+    services = data_api_client.find_services(
+        supplier_id=supplier_id,
+        framework=remove_services_for_framework,
+        status='published'
+    )['services']
+    if not services:
+        abort(400, 'Invalid framework')
```

Comment:
Same here - is this the right error message? 'No published services for this framework'?

Reply:
Updated these to be a bit more descriptive